sdk/keyvault/Azure.Security.KeyVault.Keys/src/RecoverDeletedKeyOperation.cs

```diff
@@ -5,34 +5,37 @@
 using System.Threading;
 using System.Threading.Tasks;
 using Azure.Core;
-using Azure.Core.Pipeline;
 
 namespace Azure.Security.KeyVault.Keys
 {
     /// <summary>
     /// A long-running operation for <see cref="KeyClient.StartRecoverDeletedKey(string, CancellationToken)"/> or <see cref="KeyClient.StartRecoverDeletedKeyAsync(string, CancellationToken)"/>.
     /// </summary>
-    public class RecoverDeletedKeyOperation : Operation<KeyVaultKey>
+    public class RecoverDeletedKeyOperation : Operation<KeyVaultKey>, IOperation<KeyVaultKey>
     {
         private static readonly TimeSpan s_defaultPollingInterval = TimeSpan.FromSeconds(2);
 
+        private readonly OperationInternal<KeyVaultKey> _operationInternal;
         private readonly KeyVaultPipeline _pipeline;
         private readonly KeyVaultKey _value;
-        private Response _response;
-        private bool _completed;
 
         internal RecoverDeletedKeyOperation(KeyVaultPipeline pipeline, Response<KeyVaultKey> response)
         {
             _pipeline = pipeline;
             _value = response.Value ?? throw new InvalidOperationException("The response does not contain a value.");
-            _response = response.GetRawResponse();
+            _operationInternal = new(pipeline.Diagnostics, this, response.GetRawResponse())
+            {
+                DefaultPollingInterval = s_defaultPollingInterval
+            };
+
+            _operationInternal.ScopeAttributes.Add("secret", Value.Name);
```

Comment:
It seems there are other operations in `KeyVault.Keys` setting the attribute name to "secret" (for example, `DeleteKeyOperation` and `RecoverDeletedKeyOperation`). I think we should create an issue to track these in case we already don't have one.

Reply:
@heaths would know for sure if this is correct.